versioned_docs/version-1.0/cadence-migration-guide/usdc-migration.mdx

```diff
@@ -0,0 +1,336 @@
+unless you hear otherwise from the project, your best bet is to try
+to unwind all your existing `FiatToken` and swap it to `USDCFlow` 
+so you can be ready when they eventually support `USDCFlow`.
+
+# USDC Migration Developer Guide
+
+If you are a developer who currently uses `FiatToken` in some way, you will likely
+have to do some work to remove the dependency and/or migrate
+your code and state to the new `USDCFlow` token contract.
+
+You will likely have to make multiple sets of changes if you want 
+this process to go as smoothly as possible. This obviously depends
+on the project and what kind of integration with USDC you already have.
+
+The first set is the pre-crescendo changes
+to allow users to pull their USDC out of your contracts, swap to the new `USDCFlow` token,
+and then re-enter your updated contract.
+Second, you'll have to update these changes to Cadence 1.0 so that they
+continue to work after Crescendo.
+
+There are several different categories of `FiatToken` dependence that
+any given project might fall into, and some may fall into multiple categories.
+This document describes each different category and how the `FiatToken`
+breakage affects the projects in each category.
+It also provides suggestions and examples for how each case can be switched
+and migrated to use the new `USDCFlow` smart contract and token.
+
+Some of the changes required are understandably difficult, so the Flow Foundation
+and its developers will always be available to assist in whatever capacity is needed
+to make this transition as technically, operationally, and financially easy as possible.
+Please reach out to a Flow team member in Discord or wherever you can communicate with
+Flow to ask for assistance with this upgrade.
+
+## Direct Import of FiatToken
+
+Some smart contracts import `FiatToken` directly with `import FiatToken from 0xb19436aae4d94622`.
+With [the FLIP](https://github.com/onflow/flips/pull/283)
+to be able to import and use broken types and values, a simple import
+of the broken `FiatToken` contract will not cause a contract to fail type checking,
+so if all your contract does with `FiatToken` is import it to use
+types from it as function arguments or return values, you can simply update
+those types to the corresponding `USDCFlow` types.
+
+This also applies to arguments and return values that are Capabilities
+that point to `FiatToken` vaults. These will also need to be updated
+to the new types.
+
+If your users are expecting to use the old `FiatToken` values with your contract,
+it is important for you to notify them that in order to keep using your contract,
+they need to convert their old `FiatToken` to `USDCFlow`
+via the public `USDCFlow.wrapFiatToken()` function before interacting anymore.
+It might also be useful to provide helpful error messages in functions and transactions
+that direct users to how they can swap their `FiatToken.Vault` for a `USDCFlow.Vault`.
```

Comment:
```suggestion
If your users are expecting to use the old `FiatToken` values with your contract,
it is important for you to notify them that in order to keep using your contract,
they need to convert their old `FiatToken` to `USDCFlow`
via the public `USDCFlow.wrapFiatToken()` function before they can continue interacting.

It might also be useful to provide helpful error messages in functions and transactions
that direct users to how they can swap their `FiatToken.Vault` for a `USDCFlow.Vault`.
```